core/src/main/java/io/questdb/cairo/DefaultCairoConfiguration.java

```diff
@@ -64,24 +65,33 @@ public class DefaultCairoConfiguration implements CairoConfiguration {
     private final CharSequence confRoot;
     private final long databaseIdHi;
     private final long databaseIdLo;
+    private final String dbRoot;
     private final LongSupplier importIDSupplier = () -> getRandom().nextPositiveLong();
+    private final String installRoot;
     private final CharSequence legacyCheckpointRoot;
-    private final String root;
     private final DefaultTelemetryConfiguration telemetryConfiguration = new DefaultTelemetryConfiguration();
     private final TextConfiguration textConfiguration;
     private final VolumeDefinitions volumeDefinitions = new VolumeDefinitions();
     private final boolean writerMixedIOEnabled;
 
-    public DefaultCairoConfiguration(CharSequence root) {
-        this.root = Chars.toString(root);
-        this.confRoot = PropServerConfiguration.rootSubdir(root, PropServerConfiguration.CONFIG_DIRECTORY);
+    public DefaultCairoConfiguration(CharSequence installRoot) {
+        this.installRoot = Chars.toString(installRoot);
+
+        // TODO: This is wrong, the `dbRoot` should be set to:
+        //       this.dbRoot = new File(this.installRoot, "db").getAbsolutePath();
+        //       This is a legacy bug would require fixing hundreds of test cases
+        //       which historically conflated the two directories.
+        //       The installRoot was introduced as part of PR #5375 and this change was left out for expediency.
+        this.dbRoot = this.installRoot;
```

Comment:
I tried, but something's gotta give sometimes.

Reply:
Without fixing this it's pretty hard to fix `DatabaseCheckpointAgent` since the current tests for it rely on the broken `DefaultCairoConfiguration`.

Reply:
Can we please avoid TODOs?
You can add an overload of the constructor that accepts 2 parameters `installDir` and `dbRootDir`. The constructor that takes one parameter will call

```Java
this(dbRootDir, null);
```

and then all the roots that depend on `installDir` will be nulls if it's created using `dbRootDir` only.

Reply:
Discussed with Alex and understood what I was missing.
Updated the code to imply the `installRoot` from the `dbRoot`.
This allowed fixing the `DatabaseCheckpointAgent` logic too.